### PR TITLE
22 change basename folder to pygsp2

### PR DESCRIPTION
**Pull request by cjotade**
Changes: 
* pygsp to pygsp2.
* Repo already in Pypi https://pypi.org/project/pygsp2/. 
* Readthedocs is not working since it takes the documentation from the master branch, we have to update the master branch before fix the documentation.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -127,7 +101,7 @@ See the guidelines for contributing in ``CONTRIBUTING.rst``.
 Acknowledgments
 ---------------
 
-The PyGSP was started in 2014 as an academic open-source project for
+The PyGSP2 was started in 2014 as an academic open-source project for
```
Comment (aweinstein) [inline README.rst:104]: Reference to the original project. We should not change this.

File: README.rst [hunk 2]
```diff
@@ -139,9 +113,9 @@ reproducibility, please cite the version you used as indexed by
...
 
-    @misc{pygsp,
-      title = {PyGSP: Graph Signal Processing in Python},
+    @misc{pygsp2,
+      title = {PyGSP2: Graph Signal Processing in Python},
       author = {Defferrard, Micha\"el and Martin, Lionel and Pena, Rodrigo and Perraudin, Nathana\"el},
       doi = {10.5281/zenodo.1003157},
-      url = {https://github.com/epfl-lts2/pygsp/},
+      url = {https://github.com/gsp-eeg/pygsp2/},
     }
```
Comment (aweinstein) [inline README.rst:121]: Reference to the original project. We should not change this reference to the original PyGSP.

File: pygsp2/data/santiago_metro_stations_connections.txt
Comment (aweinstein) [inline pygsp2/data/santiago_metro_stations_connections.txt:1]: This file should not be in the repo.

File: pygsp2/data/santiago_metro_stations_coords.geojson
Comment (aweinstein) [inline pygsp2/data/santiago_metro_stations_coords.geojson:1]: This file should not be in the repo.

File: pygsp2/graphs/_io.py [hunk 3]
```diff
@@ -475,7 +475,7 @@ def save(self, path, fmt=None, backend=None):
 
         .. _GraphML: https://en.wikipedia.org/wiki/GraphML
         .. _GML: https://en.wikipedia.org/wiki/Graph_Modelling_Language
-        .. _GEXF: https://gephi.org/gexf/format
+        .. _GEXF: https://gephi.org/gexf
```
Comment (aweinstein) [inline pygsp2/graphs/_io.py:478]: Why this change?

File: pygsp2/plotting.py [hunk 4]
```diff
@@ -415,7 +415,8 @@ def normalize(x):
         Set intercept value in G.plotting["normalize_intercept"]
         with value in [0, 1], default is .25.
         """
-        ptp = x.ptp()
+        #ptp = x.ptp()
+        ptp = np.ptp(x)
```
Comment (aweinstein) [inline pygsp2/plotting.py:419]: Why this change not related to the name change?

File: pygsp2/tests/test_utils2.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    def setUpClass(cls):
+        pass
+
+    @classmethod
+    def tearDownClass(cls):
+        pass
+
+    def test_graphs(self):
+        edgesfile='../data/santiago_metro_stations_connections.txt',
+        coordsfile='../data/santiago_metro_stations_coords.geojson'
```
Comment (aweinstein) [inline pygsp2/tests/test_utils2.py:27]: Note that this files should not be in the repo.

File: pygsp2/tests/test_utils2.py
Comment (aweinstein) [inline pygsp2/tests/test_utils2.py:1]: Why do we have changes in the tests in this PR?

File: pygsp2/utils_examples.py [hunk 6]
```diff
@@ -20,8 +25,8 @@
 import numpy as np
 
 
-def make_metro_graph(edgesfile='santiago_metro_stations_connections.txt',
-                     coordsfile='santiago_metro_stations_coords.geojson'):
+def make_metro_graph(edgesfile='../pygsp2/data/santiago_metro_stations_connections.txt',
+                     coordsfile='../pygsp2/data/santiago_metro_stations_coords.geojson'):
```
Comment (aweinstein) [inline pygsp2/utils_examples.py:29]: Why this change here?

File: setup.py [hunk 7]
```diff
@@ -5,41 +5,48 @@
 
 
 setup(
-    name='PyGSP',
-    version='0.5.1',
+    name='pygsp2',
+    version='2.0.3',
```
Comment (aweinstein) [inline setup.py:9]: How did we get this version number?

File: pygsp2/graphs/_io.py (hunk 3 above)
Comment (lcortesg) [reply pygsp2/graphs/_io.py:478]: The original URL doesn't exist anymore

File: pygsp2/plotting.py (hunk 4 above)
Comment (lcortesg) [reply pygsp2/plotting.py:419]: The .ptp() method was deprecated in favor of np.ptp()

Comment (aweinstein) [reply pygsp2/plotting.py:419]: OK. However, it is not related to the PyGSP to PyGSP2 change. Let's keep it anyway.

File: pygsp2/utils_examples.py (hunk 6 above)
Comment (lcortesg) [reply pygsp2/utils_examples.py:29]: The original function in "utils_examples.py" was hard coded/ bound to data contained in the examples folder, which was not accessible from within the function, easiest way to solve it was to move said files to the data folder of PyGSP.

File: pygsp2/tests/test_utils2.py (hunk 5 above)
Comment (lcortesg) [reply pygsp2/tests/test_utils2.py:27]: The original function in "utils_examples.py" was hard coded/ bound to data contained in the examples folder, which was not accessible from within the function, easiest way to solve it was to move said files to the data folder of PyGSP.

File: pygsp2/tests/test_utils2.py
Comment (lcortesg) [reply pygsp2/tests/test_utils2.py:1]: To incorporate the changes made in "utils_examples.py"

File: README.rst (hunk 2 above)
Comment (cjotade) [reply README.rst]: solved

File: README.rst (hunk 1 above)
Comment (cjotade) [reply README.rst]: solved

File: pygsp2/data/santiago_metro_stations_connections.txt
Comment (cjotade) [reply pygsp2/data/santiago_metro_stations_connections.txt:1]: We uploaded those files since we encounters some errors when trying to create the documentation in readthedocs, the files were not found and for simplicity we opt to upload them. What can we do, maybe download them when running the examples?

File: pygsp2/data/santiago_metro_stations_coords.geojson
Comment (cjotade) [reply pygsp2/data/santiago_metro_stations_coords.geojson:1]: https://github.com/gsp-eeg/pygsp2/pull/25#discussion_r1657621789

File: pygsp2/tests/test_utils2.py (hunk 5 above)
Comment (cjotade) [reply pygsp2/tests/test_utils2.py:27]: https://github.com/gsp-eeg/pygsp2/pull/25#discussion_r1657621789

File: CHANGELOG.rst
Comment (aweinstein) [inline CHANGELOG.rst:1]: This is part of the history of the project. We should not change the name here.

File: pygsp2/data/santiago_metro_stations_connections.txt
Comment (aweinstein) [reply pygsp2/data/santiago_metro_stations_connections.txt:1]: OK. Let's keep the fix of this problem in another PR.

File: setup.py (hunk 7 above)
Comment (aweinstein) [reply setup.py:9]: How we went from 0.5.1 to 2?

File: CHANGELOG.rst
Comment (aweinstein) [reply CHANGELOG.rst:1]: Fixed.
